registration/tests/serializers/test_family_detail_create.py

```diff
@@ -0,0 +1,96 @@
+                "last_name": "Weasley",
+                "information": {"1": "female"},
+            },
+        ]
+        self.guests_data = [
+            {
+                "first_name": "Harry",
+                "last_name": "Potter",
+                "information": {"3": "friend"},
+            }
+        ]
+
+    def test_family_detail_serializer_create(self):
+        num_families = Family.objects.all().count()
+        num_students = Student.objects.all().count()
+
+        data = dict(self.family_data)
+        data["parent"] = self.parent_data
+
+        serializer = FamilyDetailSerializer(data=data)
+        self.assertTrue(serializer.is_valid())
+        family = serializer.save()
+
+        self.assertEqual(Family.objects.all().count(), num_families + 1)
+        self.assertEqual(Student.objects.all().count(), num_students + 1)
+
+        self.assertEqual(family.parent.first_name, self.parent_data["first_name"])
+        self.assertEqual(family.parent.last_name, self.parent_data["last_name"])
+        self.assertEqual(family.parent.information, self.parent_data["information"])
+
+    def test_family_detail_serializer_create__children_guests(self):
+        num_families = Family.objects.all().count()
+        num_students = Student.objects.all().count()
+
+        data = dict(self.family_data)
+        data["parent"] = self.parent_data
+        data["children"] = self.children_data
+        data["guests"] = self.guests_data
+
+        serializer = FamilyDetailSerializer(data=data)
+        self.assertTrue(serializer.is_valid())
+        family = serializer.save()
+
+        self.assertEqual(Family.objects.all().count(), num_families + 1)
+        self.assertEqual(Student.objects.all().count(), num_students + 4)
+
+        self.assertEqual(family.parent.first_name, self.parent_data["first_name"])
+        self.assertEqual(family.parent.last_name, self.parent_data["last_name"])
+        self.assertEqual(family.parent.information, self.parent_data["information"])
+
+        self.assertEqual(family.children.count(), 2)
+
+        child1 = family.children.get(first_name=self.children_data[0]["first_name"])
+        self.assertEqual(child1.last_name, self.children_data[0]["last_name"])
+        self.assertEqual(child1.information, self.children_data[0]["information"])
+
+        child2 = family.children.get(first_name=self.children_data[1]["first_name"])
+        self.assertEqual(child2.last_name, self.children_data[1]["last_name"])
+        self.assertEqual(child2.information, self.children_data[1]["information"])
+
+        self.assertEqual(family.guests.count(), 1)
+        guest = family.guests.first()
+        self.assertEqual(guest.first_name, self.guests_data[0]["first_name"])
+        self.assertEqual(guest.last_name, self.guests_data[0]["last_name"])
+        self.assertEqual(guest.information, self.guests_data[0]["information"])
+
+    def test_family_detail_serializer_validate__no_parent(self):
```

Comment:
more just a question for me, but how does this test "no parent?" :0

Reply:
the `self.family_data` dict doesn't include any info about a parent! as a result `to_internal_value` [raises](https://github.com/uwblueprint/project-read-backend/pull/37/files/cc57de2d78491dfd15bb70d7e56e45a946914ad4#diff-02da605e456870643a8b6b9ff22d959f7292437d3d9d517568e631aca3256cd0R89) a `ValidationError`